### PR TITLE
Add OWASP Dependency Checker to the GH CI workflows

### PR DESCRIPTION
**Pull request by dlg99**
### Motivation

Detect CVEs early and make them visible to the committers

### Modifications

Added GH workflow that will fail if new CVE with level > 7 detected. The plan is to not treat it as blocking merge.
Currently some projects that will not pass the check and aren't included into the suppressions for variety of reasons are excluded.

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

This change is a trivial rework / code cleanup without any test coverage.

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API: (yes / no)
  - The schema: (yes / no / don't know)
  - The default values of configurations: (yes / no)
  - The wire protocol: (yes / no)
  - The rest endpoints: (yes / no)
  - The admin cli options: (yes / no)
  - Anything that affects deployment: (yes / no / don't know)

### Documentation

Check the box below or label this PR directly (if you have committer privilege).

Need to update docs? 

- [ ] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [x] `no-need-doc` 
  
  (Please explain why)
  
- [ ] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

**Review by lhotari — APPROVED** [review]
LGTM

File: .github/workflows/ci-owasp-dep-check.yaml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+name: CI - Misc - OWASP Dependency Check
+on:
+  pull_request:
```
Comment (eolivelli) [inline .github/workflows/ci-owasp-dep-check.yaml:22]: Why running this check on every pull request?
New dependencies are added very seldom.
If a dependency is flagged there will be much noise.
People usually add a new dependency using the latest available version.

I suggest to run this only periodically, like every day and manually on every active release branch

**Review by eolivelli — CHANGES_REQUESTED** [review]
Great stuff.
But I left one comment.

File: .github/workflows/ci-owasp-dep-check.yaml (hunk 1 above)
Comment (lhotari) [reply .github/workflows/ci-owasp-dep-check.yaml:22]: @eolivelli this is only run when one of the pom.xml files is modified.

Comment (lhotari) [reply .github/workflows/ci-owasp-dep-check.yaml:22]: in addition there's the existing job to run periodically

Comment (dlg99) [reply .github/workflows/ci-owasp-dep-check.yaml:22]: @eolivelli as Lari said, it runs only on changes in pom files, see:
```
      - name: Detect changed pom files
        id: changes
        uses: apache/pulsar-test-infra/paths-filter@master
        with:
          filters: |
            poms:
              - 'pom.xml'
              - '**/pom.xml'
```

**Review by eolivelli — APPROVED** [review]
Thanks for your clarifications

Lgtm

**Review by nicoloboschi — APPROVED** [review]
+1, great work

**Review by michaeljmarshall — APPROVED** [review]

File: .github/workflows/ci-owasp-dep-check.yaml [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    branches:
+      - branch-*
+
+env:
+  MAVEN_OPTS: -Dhttp.keepAlive=false -Dmaven.wagon.http.pool=false -Dmaven.wagon.http.retryHandler.class=standard -Dmaven.wagon.http.retryHandler.count=3
+
+jobs:
+
+  owasp-dep-check:
+    name:
```
Comment (michaeljmarshall) [inline .github/workflows/ci-owasp-dep-check.yaml:35]: Does it need a name @dlg99?

Comment (dlg99) [reply .github/workflows/ci-owasp-dep-check.yaml:35]: @michaeljmarshall I don't think so. Looking at other [workflows](https://github.com/apache/pulsar/tree/master/.github/workflows) it is blank there as well.

**Comment by hpvd** [comment]
+1

**Pull request merged**